runtime/port/sysvipc/j9shmem.c

```diff
@@ -1381,6 +1384,22 @@ j9shmem_createDir(struct J9PortLibrary* portLibrary, char* cacheDirName, uintptr
 		}
 	}
 
+	if (0 == omrsysinfo_get_env("HOME", homeDirPathBuffer, J9SH_MAXPATH)) {
```

Comment:
I also think it is better to move the change into: https://github.com/eclipse-openj9/openj9/blob/c9cb232d5837f091c6cf2b07615bee022da17698/runtime/port/sysvipc/j9sharedhelper.c#L126

You can use `j9shmem_getDir()` with flag `J9SHMEM_GETDIR_USE_USERHOME`. 
It should give you `$HOME/.cache/` now. You can add a new flag `J9SHMEM_GETDIR_DO_NOT_APPEND_HIDDENDIR` to make it return `$HOME/` only.
Then you can check if it is part of the `cacheDirName`.

If `J9SH_DIRPERM_ABSENT == permission` and `usingUserHome` is true, we should set permission to `01700` instead of `J9SH_PARENTDIRPERM(01777)`